### DISCUSSION TITLE
Format error

### DISCUSSION THREAD

**Discussion by mfaridbadr** [erichelgeson/BlueSCSI-v1#103] [Q&A]
@erichelgeson 

HI,
### **I use:**
- H.W. Ver. 1.0-c
- Firmware v1.0-20210410  and also v1.1-20211215
- SD Kingstone 16 GB
- File Name (HD00_2048-T14.hda)  and size 2 GB and also i try 1 GB , 512 M.B.

### Result in PC:
It work fine in my PC  (windows XP- and Adaptec AHA-2940UW SCSI controller)  and i can read and write on it

### Result in Special O.S. and special SCSI Controller:
I try it in special O.S. and special SCSI Controller  (Factory Machine) 
and the O.S.  can see the disk  and detect it's size  2 GB    but can't Format It   

### Needs:
I will try the new Firmware v1.1-20220404  but i need Support for the following :
- Create File More than 2 GB    EX: (4 GB)  and let the special O.S. take what he need from the file to create his Partitions
- How create more details log / debug to can see what is the exact happen when the special O.S. try to format  it  ,cause the normal log stopped in  `Finished initialization of SCSI Devices - Entering main loop.`

**Comment by erichelgeson** [comment]
Hi @mfaridbadr  - no need to ping on 3 different mediums - if someone can see and help they will.

I've never heard of "Special O.S." - could you provide any details on it?

Your file name has a sector size of 2048 - though that is not supported and will default to 512.

To create a blank 4gb file - you can use the tecniques here - I'd assume Special OS does not use HFS https://github.com/erichelgeson/BlueSCSI/wiki/Usage#custom-disk-images-advanced

To enable debug logging you must change the value here - https://github.com/erichelgeson/BlueSCSI/blob/main/src/BlueSCSI.cpp#L49 - note you'll need to build and re-flash. In the next release I will include a debug firmware you can flash via USB.

**Reply by marcelv-3** [reply]
if you want to log more in the LOG.TXT on the SD card you have to set 
#define DEBUG            2

but also have to alter finalizeFileLog(), 
add the bold lines otherwise the log file is closed and nothing will be logged after 'Entering main loop'

  **#if DEBUG < 2**
  LOG_FILE.close();
  **#endif**

@erichelgeson  maybe add this to the code

**Reply by erichelgeson** [reply]
Hey, good catch @marcelv-3 - would you like to open a Pull Request?

**Reply by erichelgeson** [reply]
I've included it here - https://github.com/erichelgeson/BlueSCSI/pull/111/commits/73fc0ed2bf15b04b1b0186864c2b997c3af3e99f
